### PR TITLE
Wire contexts to Apps controllers

### PR DESCRIPTION
**Pull request by damemi**
#### What type of PR is this?
/kind design

#### What this PR does / why we need it:
This wires a context to Apps controllers in kube-controller-manager, so that KCM can eventually release on cancel.

This originally came from PR https://github.com/kubernetes/kubernetes/pull/101379, which builds on top of https://github.com/kubernetes/kubernetes/pull/101125

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/daemon/daemon_controller.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ type DaemonSetsController struct {
 	burstReplicas int
```
Comment (soltysh) [inline pkg/controller/daemon/daemon_controller.go:94]: there are still a few `context.TODO` in pkg/controller/daemon/update.go and in this file too

File: pkg/controller/deployment/deployment_controller.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ type DeploymentController struct {
 	eventRecorder record.EventRecorder
 
 	// To allow injection of syncDeployment for testing.
-	syncHandler func(dKey string) error
+	syncHandler func(ctx context.Context, dKey string) error
```
Comment (soltysh) [inline pkg/controller/deployment/deployment_controller.go:75]: still a few `context.TODO` in the following files:
pkg/controller/deployment/progress.go
pkg/controller/deployment/util/deployment_util.go
pkg/controller/deployment/rollback.go
pkg/controller/deployment/sync.go

File: pkg/controller/statefulset/stateful_set.go [hunk 3]
```diff
@@ -141,22 +141,22 @@ func NewStatefulSetController(
 }
```
Comment (soltysh) [inline pkg/controller/statefulset/stateful_set.go:141]: a few more in:
pkg/controller/statefulset/stateful_set_status_updater.go
pkg/controller/statefulset/stateful_pod_control.go

**Review by soltysh — CHANGES_REQUESTED** [review]
I found still a few `context.TODO()` instances I've listed above.

/assign
/triage accepted
/priority backlog

**Comment by spiffxp** [comment]
/test pull-kubernetes-node-e2e-containerd
ref: #105381

**Comment by damemi** [comment]
/retest
Sorry for all the test failure spam, I had some trouble finding all of the missing updates locally.

**Review by soltysh — APPROVED** [review]
A few more nits, but we can handle them in a followups.
/lgtm
/approve

**Comment by alculquicondor** [comment]
Uhm.. kind of tricky, as batch is under sig-apps, but it's listed separately. Maybe you can say core apps (the core API). But it's a nit.

**Comment by nikhita** [comment]
/kind feature
/remove-kind design

We are removing the `kind/design` label. Please see https://github.com/kubernetes/community/issues/5641 for more details.

**Review by soltysh — CHANGES_REQUESTED** [review]

**Review by soltysh — CHANGES_REQUESTED** [review]
One more nit about `adopt` function in `pkg/controller/controller_ref_manager.go` and what about that `wait.UntilWithContext` in `pkg/controller/daemon/daemon_controller.go`

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**